index.js

```diff
@@ -8,15 +8,22 @@ const parse = require('module-details-from-path')
 
 module.exports = Hook
 
-const builtins = Module.builtinModules
-
-const isCore = builtins
-  ? (filename) => builtins.includes(filename)
-  // Fallback in case `builtins` isn't available in the current Node.js
-  // version. This isn't as acurate, as some core modules contain slashes, but
-  // all modern versions of Node.js supports `buildins`, so it shouldn't affect
-  // many people.
-  : (filename) => filename.includes(path.sep) === false
+/**
+ * Is the given module a "core" module?
+ * https://nodejs.org/api/modules.html#core-modules
+ *
+ * @type {(moduleName: string) => boolean}
+ */
+let isCore
+if (Module.isBuiltin) { // as of node v18.6.0
+  isCore = Module.isBuiltin
+} else {
+  isCore = moduleName => {
+    // Prefer `resolve.core` lookup to `resolve.isCore(moduleName)` because the
+    // latter is doing version range matches for every call.
+    return !!resolve.core[moduleName]
```

Comment:
REVIEW NOTE: The `resolve` module has a `core` property that looks like this:

```
> process.version
'v16.16.0'
> resolve.core
{
  assert: true,
  'node:assert': true,
  'assert/strict': true,
  'node:assert/strict': true,
  async_hooks: true,
  'node:async_hooks': true,
...
  'node:test': false,
...
```

Those true/false values are resolved from node version ranges against the current `process.version`. For example `node:test` is not a core module in node v16.16.0 that I was running this with. This makes `resolve.core` a fast way to look up if a given module id is core for the current node version. This obviates the need for https://github.com/elastic/require-in-the-middle/pull/42. (@harttle sorry that your PR for this never got merged.) It is also available back to the supported node v6, so we don't need the `filename.includes(path.sep)` fallback.

From https://nodejs.org/api/modules.html#core-modules

> The list of core modules that can be loaded without using the node: prefix is exposed as [module.builtinModules](https://nodejs.org/api/module.html#modulebuiltinmodules).

In node v18, `module.builtinModules` does *not* include `node:test`. So, our replacing usage of `module.builtinModules` handles this case as well.